supervisor/shared/external_flash/external_flash.c

```diff
@@ -121,42 +128,55 @@ static bool write_flash(uint32_t address, const uint8_t* data, uint32_t data_len
 static bool page_erased(uint32_t sector_address) {
     // Check the first few bytes to catch the common case where there is data
     // without using a bunch of memory.
-    uint8_t short_buffer[4];
-    if (read_flash(sector_address, short_buffer, 4)) {
-        for (uint16_t i = 0; i < 4; i++) {
-            if (short_buffer[i] != 0xff) {
-                return false;
+    if (flash_device->no_erase_cmd){
+        // skip this if device doesn't have an erase command.
+        return true;
+    } else {
```

Comment:
I'd prefer to remove the else and just rely on the return. That way everything inside the else is indented less. Similar below.

Reply:
Makes perfect sense. Fixed four instances of this.